File: EngineerProject.Web/src/app/services/data-services/group.service.ts
import { Post } from './../../models/Post';
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { environment } from 'src/environments/environment';
import { GroupDetails } from 'src/app/models/GroupDetails';
import { GroupAdminDetails } from 'src/app/models/GroupAdminDetails';

@Injectable({
  providedIn: 'root'
})
export class GroupService {

  constructor(private httpClient: HttpClient) { }

  getPosts(id: number) {
    const url = `${environment.apiUrl}posts/get?groupId=${id}&page=1&pageSize=10`;

    return this.httpClient.get<Post[]>(url);
  }

  getDetails(id: number) {
    const url = `${environment.apiUrl}groups/Details?id=${id}`;

    return this.httpClient.get<GroupDetails>(url);
  }

  addPost(groupId: number, title: string, content: string) {
    const url = `${environment.apiUrl}posts/Create`;

    return this.httpClient.post<any>(url, { groupId, title, content });
  }

  deletePost(id: number) {
    const url = `${environment.apiUrl}posts/Delete?id=${id}`;

    return this.httpClient.delete<any>(url);
  }

  deleteComment(id: number) {
    const url = `${environment.apiUrl}comments/Delete?id=${id}`;

    return this.httpClient.delete<any>(url);
  }

  addComment(id: number, content: string) {
    const url = `${environment.apiUrl}comments/Create`;

    return this.httpClient.post<any>(url, { postId: id, content });
  }

  modifyPost(postId: number, title: string, content: string) {
    const url = `${environment.apiUrl}posts/Modify?id=${postId}`;

    return this.httpClient.put<any>(url, { title, content });
  }

  getAdminGroupDetails(groupId: number) {
    const url = `${environment.apiUrl}groups/GetAdminGroupDetails?id=${groupId}`;

    return this.httpClient.get<GroupAdminDetails>(url);
  }

  resolveApplication(groupId: number, userId: number, accepted: boolean) {
    const url = `${environment.apiUrl}groups/ResolveApplication`;

    return this.httpClient.post<any>(url, { groupId, userId, accepted });
  }

  inviteUser(groupId: number, userIdentifier: string) {
    const url = `${environment.apiUrl}groups/InviteUser`;

    return this.httpClient.post<any>(url, { groupId, userIdentifier });
  }
}

File: EngineerProject.Mobile/Services/GroupService.cs
using EngineerProject.Commons.Dtos;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace EngineerProject.Mobile.Services
{
    public class GroupService : BaseService
    {
        public async Task<DataRequestResponse<CommentDto>> AddComment(int postId, string content)
        {
            var response = await client.PostAsync<CommentDto>($"Comments/Create", new { postId, content });

            return MapResponse(response);
        }

        public async Task<RequestResponse> AddPost(int groupId, string content)
        {
            var response = await client.PostAsync($"Posts/Create", new { groupId, content });

            return MapResponse(response);
        }

        public async Task<RequestResponse> DeleteComment(int commentId)
        {
            var response = await client.DeleteAsync($"Comments/Delete?id={commentId}");

            return MapResponse(response);
        }

        public async Task<RequestResponse> DeletePost(int postId)
        {
            var response = await client.DeleteAsync($"Posts/Delete?id={postId}");

            return MapResponse(response);
        }

        public async Task<DataRequestResponse<GroupAdminDetailsDto>> GetAdminGroupDetails(int groupId)
        {
            var response = await client.GetAsync<GroupAdminDetailsDto>($"Groups/GetAdminGroupDetails?id={groupId}");

            return MapResponse(response);
        }

        public async Task<DataRequestResponse<GroupDetailsDto>> GetDetails(int groupId)
        {
            var response = await client.GetAsync<GroupDetailsDto>($"Groups/Details?id={groupId}");

            return MapResponse(response);
        }

        public async Task<DataRequestResponse<List<MessageDto>>> GetMessages(int groupId, int page, int pageSize)
        {
            var response = await client.GetAsync<List<MessageDto>>($"Messages/Get?groupId={groupId}&page={page}&pageSize={pageSize}");

            return MapResponse(response);
        }

        public async Task<DataRequestResponse<List<PostDto>>> GetPosts(int groupId, int page, int pageSize)
        {
            var response = await client.GetAsync<List<PostDto>>($"Posts/Get?groupId={groupId}&page={page}&pageSize={pageSize}");

            return MapResponse(response);
        }

        public async Task<RequestResponse> InviteUser(int groupId, string userIdentifier)
        {
            var response = await client.PostAsync($"Groups/InviteUser", new { groupId, userIdentifier });

            return MapResponse(response);
        }

        public async Task<DataRequestResponse<List<MessageDto>>> LoadComments(int groupId, int page, int pageSize)
        {
            var response = await client.GetAsync<List<MessageDto>>($"Messages/Get?groupId={groupId}&page={page}&pageSize={pageSize}");

            return MapResponse(response);
        }

        public async Task<RequestResponse> ModifyGroup(int groupId, string name, string description, bool isPrivate)
        {
            var response = await client.PutAsync($"Groups/Modify", new { id = groupId, name, description, isPrivate });

            return MapResponse(response);
        }

        public async Task<RequestResponse> ModifyPost(int postId, string content)
        {
            var response = await client.PutAsync($"Posts/Modify?id={postId}", new { content });

            return MapResponse(response);
        }

        public async Task<RequestResponse> ResolveApplication(int groupId, int userId, bool accepted)
        {
            var response = await client.PostAsync($"Groups/ResolveApplication", new { groupId, userId, accepted });

            return MapResponse(response);
        }
    }
}